lib/child-writer/index.js

```diff
@@ -113,25 +113,36 @@ exports.write = (image, drive, options) => {
     emitter.emit('error', error);
   };
 
-  ipc.server.on('error', emitError);
-  ipc.server.on('message', (data) => {
-    let message = null;
-
+  /**
+   * @summary Bridge robot message to the child writer caller
+   * @function
+   * @private
+   *
+   * @param {String} message - robot message
+   *
+   * @example
+   * bridgeRobotMessage(robot.buildErrorMessage(new Error('foo')));
+   */
+  const bridgeRobotMessage = (message) => {
     try {
-      message = robot.parseMessage(data);
-    } catch (error) {
-      return emitError(error);
-    }
+      const parsedMessage = robot.parseMessage(message);
+
+      // The error object is decomposed by the CLI for serialisation
+      // purposes. We compose it back to an `Error` here in order
+      // to provide better encapsulation.
+      if (robot.getCommand(parsedMessage) === 'error') {
+        emitError(robot.recomposeErrorMessage(parsedMessage));
+      } else {
+        emitter.emit(robot.getCommand(parsedMessage), robot.getData(parsedMessage));
```

Comment:
This is basically the same as:

```js
emitter.emit(parsedMessage.command, parsedMessage.data);
```

with the exception that we use function accessors instead of referencing the property directly (so we can change the structure of a robot object at will without having to change code all over the place), so it doesn't look like a weird pattern IMHO.

I'm still happy to refactor if you think what would make things cleaner. What do you think?